stackrox/stackrox
# Ensure Role and Access Scope migration to postgres and UUIDs also covers references from consumers - 3.73

## Description

Backport of https://github.com/stackrox/stackrox/pull/4681 to 3.73 release.

## Checklist
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should cover the most relevant part.

In order to ensure things work properly, manual migration tests will have to cover the creation of dedicated access scopes, as well as the creation of report configurations that reference both default and manually created scopes. The generated reports before and after the migration should point at the same data.


- spawn 3.73.0 demo stackrox instance
- create access scope limited to cluster production and namespace operations
- create vulnerability report with the above scope
- edit the central container image to the one built by this PR
- spawn a central-db instance for the deployed stackrox (roxctl central db generate k8s pvc)
- edit the central container environment to enable postgres mode (env ROX_POSTGRES_DATASTORE set to true)
- let central restart
- re-open the vulnerability report
- see that the report is configured for a collection created from the above scope
